finalfusion-ffi/src/lib.rs

```diff
@@ -65,3 +71,42 @@ pub unsafe extern "C" fn ff_free_embeddings(embeddings: *mut Embeddings<VocabWra
         Box::from_raw(embeddings);
     }
 }
+
+/// Return the embedding dimensionality.
+#[no_mangle]
+pub unsafe extern "C" fn dims(embeddings: *const Embeddings<VocabWrap, StorageWrap>) -> usize {
```

Comment:
Rename to `ff_embeddings_dims`. C has no namespaces, so calling this `dims` means that this clashes with any other symbol named `dims` in another library.

Reply:
Thanks! I should've realized that last night when I couldn't call the variable `dims` in the test because `dims` was the name of the fn.